dashboard/app/views/levels/_teacher_panel.html.haml

```diff
@@ -3,5 +3,36 @@
 - data[:script_id] = @script.id
 - data[:section] = @section&.summarize
 
+- if @level && @script_level
+  - if @level.try(:examples).present? && current_user.authorized_teacher? # 'solutions' for applab-type levels
+    - level_example_links = @level.examples.map do |example|
+      - send("#{@level.game.app}_project_view_projects_url".to_sym, channel_id: example, host: 'studio.code.org', port: 443, protocol: :https)
+  - if @level.ideal_level_source_id && @script # 'solutions' for blockly-type levels
+    -level_example_links = []
+    -level_example_links.push(build_script_level_url(@script_level, {solution: true}.merge(@section ? {section_id: @section.id} : {})))
+
+- data[:level_examples] = level_example_links
+
+- data[:section_levels] = @section.students.order(:name).map do |student|
```

Comment:
Nice job figuring out all of what you need to send down from the server to get this working! This entire assignment to `section_levels` is a bit hard to understand as it is, though, and it's a bit too much logic for a view. As a first step, I'd recommend  extracted this logic to `def summarize_for_teacher_panel(student)` in script_level.rb, similar to summarize_as_bonus in that file, or similar to `summary_for_lesson_plans` in level.rb. this should make it possible to add some unit tests for this logic (could be in a next PR).

Reply:
Also can we call these `section_script_levels` instead of `section_levels`? If these really are more like levels than script levels, let's discuss.

Reply:
Thanks for this suggestion Dave! I was really feeling like this was getting ugly but I had no idea how to make it better. I will work on this!